### DISCUSSION TITLE
Mutate in useQuery

### DISCUSSION THREAD

**Discussion by bpCodes** [TanStack/query#3372] [Q&A] [answered]
If i have multiple useQuery instances when I make a mutation in onSuccess it fetches multiple times

Example 
```
const { status, data, isFetching, error, failureCount, refetch } = useQuery(
    ["todos", { filter }],
    () => fetchTodos({ filter }),
    {
      onSuccess: (data) => {
        if(data.length === 0){
          return finishTask.mutate({ taskName: "Some" });  
        }
      }
    }
  );
```

`.mutate` sends multiple request 

Why is this happening?

**Comment by TkDodo** [comment, marked as answer]
Because onSuccess is called for every useQuery instance.

**Reply by bpCodes** [reply]
is there a recommended way to make dependant mutation? Because the only way I think is 
```
  const { data } = useQuery());

  React.useEffect(() => {
    if(data && data.length === 0){
      finishTask.mutate()
    }
  }, [data])

```

Thanks for response

**Reply by TkDodo** [reply]
the effect will also run once for every component that has the effect, so it's pretty much the same as using `onSuccess`. Only difference is that `onSuccess` is not called if data is only read from the cache, e.g. if you have a `staleTime` set.

The [global cache callbacks](https://react-query.tanstack.com/reference/QueryCache#global-callbacks) provide a way to perform side effects globally once.

You could also just run a second request in side the `queryFn`.
